node/pkg/logscribe/app.go

```diff
@@ -87,3 +148,125 @@ func bulkCopyLogs(ctx context.Context, logsChannel <-chan *[]api.LogInsertModel)
 		}
 	}
 }
+
+func ProcessLogs(ctx context.Context) map[string][]LogInsertModelWithIDWithCount {
```

Comment:
the function seems too much complicated too me, I'd like to suggest a refactoring.

can this function be modularized so that it calls the process func per each services?

the map types at the beginning also seems to be complicated rather than generating map to be processed in a whole, let's just send results per service to the createGithubIssue func

Reply:
and please modularize the slicing func that filters out top occurences